templates/components/engines/python/chat/__init__.py

```diff
@@ -16,9 +37,13 @@ def get_chat_engine(filters=None, params=None):
             ),
         )
 
-    return index.as_chat_engine(
-        similarity_top_k=int(top_k),
-        system_prompt=system_prompt,
-        chat_mode="condense_plus_context",
+    retriever = index.as_retriever(
+        similarity_top_k=top_k,
         filters=filters,
     )
+
+    return CondensePlusContextChatEngine.from_defaults(
+        system_prompt=system_prompt,
```

Comment:
how about we have an env variable like:

`RENDER_CITATIONS` if it is `true`, then we append the system prompt
AND add the `NodeCitationProcessor` (because it needs this system prompt)

Reply:
or we use `SYSTEM_CITATION_PROMPT` which is defined as in `env-variable.ts` only if it is defined, we add it here to the system prompt and then we add the node processor